lib/src/model/puzzle/puzzle_providers.dart

```diff
@@ -73,3 +80,74 @@ Future<PuzzleDashboard> puzzleDashboard(
   }
   return result.asFuture;
 }
+
+@riverpod
+class PuzzleHistory extends _$PuzzleHistory {
+  final List<HistoryPuzzle> _list = [];
+  StreamSubscription<HistoryPuzzle>? _streamSub;
+  PuzzleRepository? _repo;
+  DateTime? _lastDateAfterFirstFetch;
+  DateTime? _lastDate;
+
+  @override
+  Future<PuzzleHistoryState> build() async {
+    ref.cacheFor(const Duration(seconds: 30));
+    ref.onDispose(() => _streamSub?.cancel());
+
+    _repo = ref.watch(puzzleRepositoryProvider);
+    final stream =
+        _repo!.puzzleActivity(10, DateTime.now()).asBroadcastStream();
+    _streamSub = stream.listen(
+      (event) {
+        _list.add(event);
+      },
+      onDone: () {
```

Comment:
so here the stream is not used as a stream, rather a future. So we could simplify a lot by removing completely this class and return directly a `Future<List<HistoryPuzzle>>` from the `PuzzleRepository` (see comment here)

Reply:
I guess I over engineered the solution while trying to fix some other problems  😆